tests/test_api.py

```diff
@@ -85,3 +91,111 @@ def test_qasm_to_phir(self) -> None:
         """
 
         assert qasm_to_phir(qasm, QtmMachine.H1_1)
+
+    def test_qasm_to_phir_with_wasm(self) -> None:
+        """Test the qasm string entrypoint works with WASM."""
+        qasm = """
+        OPENQASM 2.0;
+        include "qelib1.inc";
+
+        qreg q[2];
+        h q;
+        ZZ q[1],q[0];
+        creg cr[3];
+        creg cs[3];
+        creg co[3];
+        measure q[0]->cr[0];
+        measure q[1]->cr[1];
+
+        cs = cr;
+        co = add(cr, cs);
+        """
+
+        wasm_bytes = get_wat_as_wasm_bytes(WatFile.add)
+
+        wasm_uid = hashlib.sha256(base64.b64encode(wasm_bytes)).hexdigest()
+
+        phir_str = qasm_to_phir(qasm, QtmMachine.H1_1, wasm_bytes=wasm_bytes)
+        phir = json.loads(phir_str)
+
+        expected_metadata = {"ff_object": (f"WASM module uid: {wasm_uid}")}
+
+        assert phir["ops"][21] == {
+            "metadata": expected_metadata,
+            "cop": "ffcall",
+            "function": "add",
+            "args": ["cr", "cs"],
+            "returns": ["co"],
+        }
+
+    def test_pytket_with_wasm(self) -> None:
+        wasm_bytes = get_wat_as_wasm_bytes(WatFile.testfile)
+        phir_str: str
+        try:
+            wasm_file = NamedTemporaryFile(suffix=".wasm", delete=False)
+            wasm_file.write(wasm_bytes)
+            wasm_file.flush()
+            wasm_file.close()
+
+            w = WasmFileHandler(wasm_file.name)
+
+            c = Circuit(6, 6)
+            c0 = c.add_c_register("c0", 3)
+            c1 = c.add_c_register("c1", 4)
+            c2 = c.add_c_register("c2", 5)
+
+            c.add_wasm_to_reg("multi", w, [c0, c1], [c2])
+            c.add_wasm_to_reg("add_one", w, [c2], [c2])
+            c.add_wasm_to_reg("no_return", w, [c2], [])
+            c.add_wasm_to_reg("no_parameters", w, [], [c2])
+
+            c.add_wasm_to_reg("add_one", w, [c0], [c0], condition=c1[0])
+
+            phir_str = pytket_to_phir(c, QtmMachine.H1_1)
+        finally:
+            Path.unlink(Path(wasm_file.name))
+
+        phir = json.loads(phir_str)
+
+        expected_metadata = {"ff_object": (f"WASM module uid: {w!s}")}
```

Comment:
Here, since we are in pytket land, we can just use the string representation of `WasmFileHandler` instance.